tests/plugins/clipboard/mobilefixcontextmenu.js

```diff
@@ -0,0 +1,119 @@
+/* bender-tags: editor,clipboard */
+/* bender-ckeditor-plugins: toolbar,clipboard,wysiwygarea,contextmenu */
+
+( function() {
+	'use strict';
+
+	bender.editors = {
+		classic: {
+			config: {
+				language: 'en'
+			}
+		},
+
+		divarea: {
+			config: {
+				language: 'en',
+				extraPlugins: 'divarea'
+			}
+		},
+
+		inline: {
+			creator: 'inline',
+			config: {
+				language: 'en'
+			}
+		}
+	};
+
+	function assertTouchEnd( bot, editor ) {
+		var notificationSpy = sinon.spy( editor, 'showNotification' ),
+			dialogSpy = sinon.spy(),
+			dialogListener;
+
+		dialogListener = editor.on( 'pasteDialog', dialogSpy );
+
+		editor.once( 'dialogShow', function( evt ) {
+			resume( function() {
+				var dialog = evt.data;
+
+				dialogListener.removeListener();
+
+				assert.areSame( 1, dialogSpy.callCount, 'pasteDialog event count' );
+
+				dialog.fire( 'cancel' );
+				dialog.hide();
+			} );
+		} );
+
+		bot.contextmenu( function( menu ) {
+			var menuItem = menu.findItemByCommandName( 'paste' );
+
+			menuItem.element.once( 'touchend', function() {
+				assert.isTrue( editor._.forcePasteDialog, 'Forcing paste dialog' );
+				assert.isFalse( notificationSpy.called, 'Notification not shown' );
+				notificationSpy.restore();
+			}, null, null, 999 );
+
+			menuItem.element.fire( 'touchend' );
+			if ( CKEDITOR.env.edge ) {
+				// Trigger Paste command directly as button click does not trigger it on Edge.
+				editor.execCommand( 'paste' );
+			}
+			menuItem.element.$.click();
+
+			wait();
+		} );
+	}
+
+	function assertClick( bot, editor ) {
+		var notificationSpy = sinon.spy( editor, 'showNotification' );
+
+		bot.contextmenu( function( menu ) {
+			var menuItem = menu.findItemByCommandName( 'paste' );
+
+			menuItem.element.once( 'click', function() {
+				assert.isFalse( editor._.forcePasteDialog, 'Forcing paste dialog' );
+				if ( !CKEDITOR.env.ie || CKEDITOR.env.edge ) {
+					// There are no paste notifications on IE browsers.
+					assert.areSame( 1, notificationSpy.callCount, 'Notification shown' );
+				}
+				notificationSpy.restore();
+			}, null, null, 999 );
+
+			if ( CKEDITOR.env.edge ) {
+				// Trigger Paste command directly as button click does not trigger it on Edge.
+				editor.execCommand( 'paste' );
+			}
+			menuItem.element.$.click();
```

Comment:
If it doesn't trigger on Edge, it should be in `else` condition, right?

Reply:
The problem here is that `click` on the context menu item does not trigger pasting in Edge (so it needs to be triggered manually), however the [test listens on `click` event](https://github.com/ckeditor/ckeditor-dev/pull/1355/files#diff-101bd29e15b6f58a30b0f693fbb35e4bR75) and proceeds there with asserting things so it is still needed for all browsers.